src/VisualStudio/CSharp/Impl/EventHookup/EventHookupSessionManager.cs

```diff
@@ -29,14 +38,34 @@ internal void EventHookupFoundInSession(EventHookupSession analyzedSession)
 
             var caretPoint = analyzedSession.TextView.GetCaretPoint(analyzedSession.SubjectBuffer);
 
+            // only generate tooltip if it is not already shown (_toolTipPresenter == null)
             // Ensure the analyzed session matches the current session and that the caret is still
             // in the session's tracking span.
-            if (CurrentSession == analyzedSession &&
+            if (_toolTipPresenter == null &&
+                CurrentSession == analyzedSession &&
                 caretPoint.HasValue &&
                 analyzedSession.TrackingSpan.GetSpan(CurrentSession.TextView.TextSnapshot).Contains(caretPoint.Value))
             {
-                // Watch all text buffer changes & caret moves while this quick info session is
-                // active
+                // Create a tooltip presenter that stays alive, even when the user types, without tracking the mouse.
+                _toolTipPresenter = this._toolTipService.CreatePresenter(analyzedSession.TextView,
+                    new ToolTipParameters(trackMouse: false, ignoreBufferChange: true));
+
+                // tooltips text is: Program_MyEvents;      (Press TAB to insert)
+                // GetEventNameTask() gets back the event name, only needs to add a semicolon after it.
+                var eventText = analyzedSession.GetEventNameTask.Result + ";";
+                var texts = new[] { eventText, CSharpEditorResources.Press_TAB_to_insert };
+                var textRuns = texts.Select(s => new ClassifiedTextRun(ClassificationTypeNames.Text, s));
+                var content = new[] { new ClassifiedTextElement(textRuns) };
+
+                // TODO: will file a bug for this
+                // use string[] for now to unblock tests for now, otherwise editor throws ArgumentNullException when running unit tests
+                // might be a MEF issue
```

Comment:
I think this investigation should still be investigated prior to merge here? Otherwise odd we spend lots of time making content to then ignore it per a bug that's still potentially our fault?